### PR TITLE
tests: add tar source integration test (CRAFT-197)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Make sure we can correctly read a valid tar source.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: tests/integration/sources/test_tar.py [hunk 1]
```diff
@@ -1 +1,75 @@
...
+        """\
+        parts:
+          foo:
+            plugin: make
+            source: foobar.tar.gz
+        """
+    )
+
+    parts = yaml.safe_load(_parts_yaml)
+    src = _LOCAL_DIR / "data" / "foobar.tar.gz"
```
Comment (cjp256) [inline tests/integration/sources/test_tar.py:42]: Might be nicer to create a tar as part of the test (or as fixture) rather than track the blob

**Review by cjp256 — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]
I agree with Chris, this might also be a good reason to ammend the commit instead of pushing a new one :-)

File: tests/integration/sources/test_tar.py (hunk 1 above)
Comment (cmatsuoka) [reply tests/integration/sources/test_tar.py]: Changed to create the tar file programmatically in the test.

**Comment by cmatsuoka** [comment]
Tar file created programmatically in the test instead of using a binary asset.

**Pull request merged**